### PR TITLE
Fix error recovery for generic EventCallback nodes

### PR DESCRIPTION
**Pull request by 333fred**
During error recovery when rewriting generic Components, we would blindly dereference the dictionary of bindings, even though it was not supplied for this path. This was causing a high-hit watson in VS4Mac, as well as a relatively lower-hit watson in regular VS. I nullable-annotated the rewriting pass, which immediately showed the unsafe dereference.

Fixes (internal links) https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1558766, https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1564376.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/razor-compiler for review.

File: src/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs [hunk 1]
```diff
@@ -257,7 +257,7 @@ private void Process(ComponentIntermediateNode node)
                 // helpful as possible. So let's substitute 'object' for all of those type parameters, and add
                 // an error.
                 var mappings = bindings.ToDictionary(kvp => kvp.Key, kvp => kvp.Value.Content);
-                RewriteTypeNames(_pass.TypeNameFeature.CreateGenericTypeRewriter(mappings), node);
+                RewriteTypeNames(_pass.TypeNameFeature.CreateGenericTypeRewriter(mappings), node, bindings: bindings);
```
Comment (333fred) [inline src/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs:260]: This is the actual bugfix.

**Review by chsienki — APPROVED** [review]

**Comment by 333fred** [comment]
@dotnet/razor-compiler for a second review please.

**Comment by 333fred** [comment]
@dotnet/razor-compiler for a second review please.

**Pull request merged**